File: javascript/eliuds-eggs/eliuds-eggs.js
export const eggCount = (displayValue) => {
  /* iteratively: */
  let count = 0;
  while (displayValue > 0) {
    count += displayValue & 1;
    displayValue >>= 1;
  }
  return count;

  /* functionally 
  return displayValue
    .toString(2)
    .split('')
    .filter((digit) => digit == "1")
    .length;
  */
};

File: julia/eliuds-eggs/eliuds-eggs.jl
#= iterative
function eggcount(number)
    count = 0
    while number > 0
        count += number & 1
        number >>= 1
    end
    count
end
=#

# recursive
function eggcount(number, count=0)
    if number <= 0
        return count
    end
    return eggcount(number >> 1, count + (number & 1))
end
